### DISCUSSION TITLE
Using Controller component throws warning

### DISCUSSION THREAD

**Discussion by emilpaw** [react-hook-form/react-hook-form#4918] [Q&A] [answered]
I am currently struggling to integrate React-Hook-Form with Semantic UI React. Here is a MVP of a somewhat working Input Field: https://codesandbox.io/s/infallible-chandrasekhar-pqbi3?file=/src/App.js
Is this the right way to do this? It throws following warning:

Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components

Additionally, in what way could I also add the not yet registered Select Box? I have troubles wrapping my head around using RHF with Component libraries. It's a breeze for native elements though. Thanks for your work on it.

**Comment by bluebill1049** [comment, marked as answer]
while I hate to say this, but please read the doc.

<img width="970" alt="Screen Shot 2021-04-23 at 10 01 42 am" src="https://user-images.githubusercontent.com/10513364/115799015-e8fe0a80-a41a-11eb-86c6-cdb7309f9f05.png">

You need to either set defaultValue at the field-level or call useForm with defaultValues.

it's throwing error because of that.

**Reply by emilpaw** [reply]
Thanks for the fast response.
I misunderstood the defaultValues as being optional.

With providing the defaultValues to useForm it gives me following warning though (I changed the codesandbox from above):
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

**Reply by justgowa** [reply]
It's because field contains ref) 
Check sandbox below.
https://codesandbox.io/s/eloquent-leftpad-w90zo

**Reply by emilpaw** [reply]
@justgowa Thanks. I exluded the ref from the Dropdown too but still no luck with that. Any suggestion?
https://codesandbox.io/s/wizardly-water-p4fgn?file=/src/App.js

**Reply by justgowa** [reply]
@emilpaw  the error is not related to RHF.
SUI uses findDomNode API that not valid in React Strict Mode
![image](https://user-images.githubusercontent.com/46902316/115903967-a91c3f00-a47d-11eb-9c4a-50e70a86fcc9.png)

**Reply by emilpaw** [reply]
Yes I know that this is an ongoing issue with SUIR. I mean that it doesn't seem to register that the value changed after selecting something and still shows the error message.

**Reply by justgowa** [reply]
@emilpaw  Try it)
https://codesandbox.io/s/mystifying-albattani-dundp?file=/src/App.js

**Reply by emilpaw** [reply]
@justgowa Thank you very much!

**Reply by saidhasyim** [reply]
Hi, I hardcoded the defaultValue = one of the option menu value, and the error disappeared. However, I need this to be dynamic because it needs to pull the data from database when someone does an "Edit".

So, i set it as follow. It works fine in "Edit," but when someone does an "Add new" the error occurs.
`defaultValue={itemData && itemData.bookId}`

Is there a way to work around this? Thanks

**Reply by Moshyfawn** [reply]
Try setting `defaultValue` to an appropriate default value for your input and resetting your form data with the `reset` method as a side-effect, when the data arrives 
```js
useEffect(() => {
  if (data) {
    reset(data)
  }
}, [data, reset])
```

**Reply by saidhasyim** [reply]
> 

Thanks so much, Moshyfawn. It's working.
